montage/mw.py

```diff
@@ -115,18 +115,22 @@ def endpoint(self, next, cookie, rdb_session, _route, config,
                         or '/static/' in _route.pattern
                         or isinstance(_route, NullRoute))
 
-        try:
-            userid = cookie['userid']
-        except (KeyError, TypeError):
-            if config['__env__'] == 'devtest':
-                # TODO: until github.com/pallets/werkzeug/issues/1060
-                userid = 6024474
-            else:
-                if ep_is_public:
-                    return next(user=None, user_dao=None)
-                err = 'invalid cookie userid, try logging in again.'
-                response_dict['errors'].append(err)
-                return {}
+        config_userid = config.get('userid')
+        if config_userid:
+            userid = config_userid
+        else:
+            try:
+                userid = cookie['userid']
+            except (KeyError, TypeError):
+                if config['__env__'] == 'devtest':
+                    # TODO: until github.com/pallets/werkzeug/issues/1060
+                    userid = 6024474
```

Comment:
also, maybe this should be `config.get('userid', 6024474)` for a unified devtest experience?

Reply:
I have '__env__': 'dev', not 'devtest'. How should we unify that?

Reply:
ah, good question. devtest is for the pytest run. dev is for local dev. devtest only comes from `utils.py`. With this PR, we could move the userid into that dictionary in `utils.py`, and allow userid overriding for dev and devtest. 

I don't think we want to unify dev and devtest because a stray config.dev.yaml could change the pytest behavior. Mostly I want to limit the envs where `userid` is considered. Does that make sense?